Source/WebCore/dom/Document.cpp

```diff
@@ -4988,6 +4991,8 @@ bool Document::setFocusedElement(Element* element, const FocusOptions& options)
         if (options.trigger != FocusTrigger::Bindings)
             m_latestFocusTrigger = options.trigger;
 
+        newFocusedElement->updateContentRelevancyStatus(ContentRelevancyStatus::Focused, true);
```

Comment:
I think the same bug exists for focus as well, same bug as top layer where descendants wouldn't be updated.